charts/opencti/templates/connector/deployment.yaml

```diff
@@ -75,6 +75,12 @@ spec:
             {{- toYaml .securityContext | nindent 12 }}
           image: "{{- if and .image (hasKey .image "repository") }}{{ .image.repository }}{{- else if $.Values.global.imageRegistry }}{{ printf "%s/opencti/%s" $.Values.global.imageRegistry .name }}{{- else }}{{ printf "opencti/%s" .name }}{{- end }}:{{ if and .image (hasKey .image "tag") }}{{ .image.tag | default $.Chart.AppVersion }}{{ else }}{{ $.Chart.AppVersion }}{{ end }}"
           imagePullPolicy: {{ if and .image (hasKey .image "pullPolicy") }}{{ .image.pullPolicy }}{{ else }}IfNotPresent{{ end }}
+          ports:
+            {{- if ((.serviceMonitor).enabled) }}
```

Comment:
Why `((.serviceMonitor).enabled` instead `.serviceMonitor.enabled`?

Reply:
Will do!

The () make sure the serviceMonitor is optional. This was my solution to ensure that if a user does not specify the servicemonitor value on a connector, the helm chart assumes the the variable is false.

If I don't add this and the user does not include the servicemonitor value it will throw a Null error.

Reply:
Hmmm, I thought parentheses in the expression `{{- if ((.serviceMonitor).enabled) }}` aren't necessary to handle optional properties in Helm. The expression `{{- if .serviceMonitor.enabled }}` is sufficient to check if the enabled property is defined and set to `true`. If `serviceMonitor` or `enabled` are not defined, the condition will evaluate to `false`, and the corresponding block will not be processed.

Reply:
Ah okay. I'll validate and get back to you :)

Reply:
If I remove parentheses from the serviceMonitor and then also remove the serviceMonitor from the connector values I get the following error:
![image](https://github.com/user-attachments/assets/f467eca2-8473-434a-86bd-a33b82d11cb6)
I can also replicate this with helm template --debug

This is not an issue for servicemonitor on the server deployment for instance, because there are default values defined in values.yaml. I believe this cannot be done for connectors as users can add any connector they want. 

https://stackoverflow.com/questions/59795596/how-to-make-nested-variables-optional-in-helm

Perhaps there is a better way of solving this?

Reply:
Ok! I'll approve with parentheses. I study to use this on the rest of code, I was unaware of this functionality.